docs/specs/servers/push/spec.md

```diff
@@ -84,27 +93,42 @@ DELETE <PUSH_SERVER_URL>/clients/<CLIENT_ID>
 
 ```
 POST <PUSH_SERVER_URL>/clients/<CLIENT_ID>
+```
+
+- `CLIENT_ID` - This is used to map to the device token
+
+Headers:
+- `X-Ed25519-Timestamp` - timestamp of the signature
+- `X-Ed25519-Signature` - hex encoded ed25519 signature of `format!("{}.{}.{}", timestamp, body.len(), body)` where `body` is the request body
+
+Request body:
+```typescript
 {
-    "id": "0000-0000-0000-0000"
-    "payload": {
-      "topic": "<topic>",
-      "flags": <flags>,
-      "blob": "<blob>"
-    }
-```
-
-- `CLIENT_ID`: This is used to map to the device token
-- `id`: This is generated by the Relay to prevent duplicate notifications
-- `payload`: A string with 3 values encoded and split with a `:` as a delimiter.
-  - `topic`: This is used by the SDKs to decrypt encrypted payloads on the client side
-  - `flags`: an integer value that uses bitmasks to encode information about the payload for the SDKs, the flags are below
-    ```
-    Encrypted  = 1 << 0
-    Sign       = 1 << 1
-    Auth       = 1 << 2
-    Chat       = 1 << 3
-    Notify     = 1 << 4
-    Encrypted_Sign = 1100 / 1108 (session_propose / session_request)
-    Encrypted_Auth = 3000 (auth_request)  
-    ```
-  - `blob`: If the payload is encrypted this is just the encrypted blob, when this is cleartext it should be base64 encoded. The blob should follow the format from of the [Notify Message](https://specs.walletconnect.com/2.0/specs/clients/notify/data-structures#notify-message)
+  // Message ID for deduplication purposes
+  // Note: When `payload.raw` is set, `sha264(payload.raw)` should be used as the Message ID instead
+  id: string,
+
+  payload: {
+    // Topic of the message. This is used by the SDKs to decrypt encrypted payloads on the client side
+    topic: string,
+
+    // Raw message sent on the topic
+    raw: string,
+
+    // See below
+    flags: number,
+
+    // See below
+    blob: string,
+  }
+```
+
+- `flags`: an integer value that uses bitmasks to encode information about the payload for the SDKs, the flags are below
+  ```
+  Encrypted  = 1 << 0
+  Sign       = 1 << 1
+  Auth       = 1 << 2
+  Chat       = 1 << 3
+  Notify     = 1 << 4
```

Comment:
```suggestion
  Encrypted  = 1 << 0
  Sign       = 1 << 1
  Auth       = 1 << 2
  Chat       = 1 << 3
  Notify     = 1 << 4
  Encrypted_Sign = 1100 / 1108 (session_propose / session_request)
  Encrypted_Auth = 3000 (auth_request) 
```

SDK would need these two new flags to know how to parse a blob

Reply:
> SDK would need these two new flags to know how to parse a blob

I don't think so because if the push client sets `always_raw: true` then the raw encrypted message would always be sent and it would always need to be decrypted.

Reply:
My point is that even though the `always_raw` flag is set to true, for SDK it would be beneficial to know the type of the request. It would be needed to know the type to which SDK should parse JSON

It has nothing to do with decryption

By sending a request tag as a notification flag we could easily achieve this

Reply:
(We agreed the JSON-RPC `method` can be used to parse the JSON)